src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4768,11 +4768,16 @@ internal object CreateInstanceSlow(bool publicOnly, bool wrapExceptions, bool sk
         {
             RuntimeMethodHandleInternal runtime_ctor = default;
             bool bCanBeCached = false;
+            bool bHasNoDefaultCtor = false;
 
             if (!skipCheckThis)
                 CreateInstanceCheckThis();
 
-            object instance = RuntimeTypeHandle.CreateInstance(this, publicOnly, wrapExceptions, ref bCanBeCached, ref runtime_ctor);
+            object instance = RuntimeTypeHandle.CreateInstance(this, publicOnly, wrapExceptions, ref bCanBeCached, ref runtime_ctor, ref bHasNoDefaultCtor);
+            if (bHasNoDefaultCtor)
+            {
+                throw new MissingMethodException("No parameterless constructor defined for " + this);
```

Comment:
This error needs to be localized. I think you may want to change the `Arg_NoDefCTor` message to include the type name - add also the few remaining places that use this to include the type name as well.

Reply:
Fixed

Reply:
I believe `Arg_NoDefCTor` is also used in [src/System.Private.CoreLib/src/System/Activator.cs](https://github.com/dotnet/coreclr/blob/cc52c67f5a0a26194c42fbd1b59e284d6727635a/src/System.Private.CoreLib/src/System/Activator.cs#L224)

Reply:
Thanks for the catch, fixed as well.